### ISSUE TITLE
[Doc] `parametrize_with_cases` takes list of such elements

### ISSUE THREAD

**Issue by eddiebergman** [smarie/python-pytest-cases#257]
Ello,

While the [documentation](https://smarie.github.io/python-pytest-cases/api_reference/#parametrize_with_cases), the type signature for `cases` doesn't seem to indicate it takes lists
of such items `Callable`, `Type` and `ModuleRef` while the description below says it does.

```python
@parametrize_with_cases(argnames: str,
                        cases: Union[Callable, Type, ModuleRef] = AUTO,
                        prefix: str = 'case_',
                        glob: str = None,
                        has_tag: Union[str, Iterable[str]] = None,
                        filter: Callable = None,
                        ids: Union[Callable, Iterable[str]] = None,
                        idstyle: Union[str, Callable] = None,
                        scope: str = "function",
                        import_fixtures: bool = False
                        )
```
> Finally, the cases argument also accepts an explicit case function, cases-containing class, module or module name; or a list of such elements. Note that both absolute and relative module names are suported.

_typo in "suported" that can be fixed :)_

I understand this would complicate the type signature but I was hoping to distuinguish between the two cases. The code seems to indicate the second option here.
```python
# cases must be a list of same type
cases: Union[Callable, Type, ModuleRef, List[Callable], List[Type], List[ModuleRef]]
# or cases can be a list of any type
cases: Union[Callable, Type, ModuleRef, List[Union[Callable, Type, ModuleRef]]]
```

In any case, do you think it makes sense to make this more explicit? I imagine making the type signature more explicit in the docs is much too long but at least in the documentation to clarify it?

**Comment by eddiebergman** [comment]
I can look into this later this week, I assume some tests to actually verify that this behaviour is correct might be required i.e. That a `CaseType` is possible or a list of any possible `Iterable[CaseType]` is possible?

I havn't looked through the tests of `pytest_cases` so it may already be there

**Comment by smarie** [comment]
Yes the tests already support it since the corresponding code is covered https://smarie.github.io/python-pytest-cases/reports/coverage/d_e0f0fa8c1f5d62a2_case_parametrizer_new_py.html#t240  (line 245)

This is really just type hints, sorry for this :'(

**Comment by eddiebergman** [comment]
Cool, thanks for pointing me to it, it should be a quick doc update then :) I'll have a look later today so!

**Comment by smarie** [comment]
Closed by #259
